third-party/realdds/src/dds-device-impl.cpp

```diff
@@ -256,33 +258,68 @@ void dds_device::impl::on_option_value( nlohmann::json const & j, eprosima::fast
     {
         throw std::runtime_error( "control is not an object" );
     }
-    std::string option_name;
-    if( ! rsutils::json::get_ex( control, option_name_key, &option_name ) )
-    {
-        throw std::runtime_error( "missing control/option-name" );
-    }
-    
-    // Find the option and set its value
+
+    // Find the relevant (stream) options to update
     dds_options const * options = &_options;
     std::string stream_name;  // default = empty = device option
-    if( rsutils::json::get_ex( control, stream_name_key, &stream_name ) && !stream_name.empty() )
+    if( rsutils::json::get_ex( control, stream_name_key, &stream_name ) && ! stream_name.empty() )
     {
         auto stream_it = _streams.find( stream_name );
         if( stream_it == _streams.end() )
+            throw std::runtime_error( "stream not found" );
+        options = &stream_it->second->options();
+    }
+
+    auto update_option = [&]( std::string const & option_name, float const new_value )
```

Comment:
I don't like naming a lambda and calling it later by name. Why not make it a regular function?

Reply:
Because making it a regular function means a whole lot more code that sits separate (and probably would need to be declared in the class header) from the function where it's being used.
If we need to refactor or reuse this code later on, I agree. But for now this is self-contained.